pydantic/config.py

```diff
@@ -33,6 +33,9 @@ class ConfigDict(TypedDict, total=False):
     title: str | None
     """The title for the generated JSON schema, defaults to the model's name"""
 
+    model_name: str | None
+    """If not `None` overrides the model names generated by `pydantic.v1.schema.get_model_name_map()`."""
```

Comment:
```suggestion
    """Name assigned to `__name__` and used in the model's JSON schema"""
```

Reply:
We can omit the mentions of inner-workings of the code and instead give a more broad description here that'll help users decide if they want to set this field.